gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -114,18 +121,45 @@ private void writePolicyMetadata(final Map<String, PolicyMetadata> policyMetadat
         }
     }
 
-    private Set<Dependency> getPolicyDependencies(final String id, final Bundle rawBundle){
+    private Set<Dependency> getPolicyDependencies(final String id, final Bundle rawBundle, final Element policy) {
         Map<Dependency, List<Dependency>> dependencyListMap = rawBundle.getDependencyMap();
+        final Set<Dependency> dependencies = new HashSet<>();
         if (dependencyListMap != null) {
             Set<Map.Entry<Dependency, List<Dependency>>> entrySet = dependencyListMap.entrySet();
             for (Map.Entry<Dependency, List<Dependency>> entry : entrySet) {
                 Dependency parent = entry.getKey();
                 if (parent.getId().equals(id)) { // Search for "id" to get its dependencies
-                    return new HashSet<>(entry.getValue());
+                    List<Dependency> policyDependencies = entry.getValue();
```

Comment:
I would expect logic to be simple. Let me explain.
why do we capture the dependencies? to bundle the entity along with all its dependencies.
ENCAS and its definition policy must be bundled as single unit. Having said that, the definition policy will always be the dependent entry over ENCAS etity. 

Hence, treat the definiton policy as implicit dependency rather than explicit one.

Reply:
Fix is to remove the defining policy from the dependency list.
After calling the getPolicyDependencies() method, simply remove the policy from the set.

Reply:
@graju256, please review, made the changes as requested